From a3e76cf4df52c3ad728b85acd0fbd0436c226dea Mon Sep 17 00:00:00 2001
From: Michael-CY Lee <michael-cy.lee@mediatek.com>
Date: Wed, 4 Jun 2025 16:33:38 +0800
Subject: [PATCH 1/2] mtk: mt76: mt7915: do not set key for cipher CMAC-128

Although FW does support setting IGTK for cipher CMAC, it does not
support setting IGTK for GMAC, which requires MAC80211 to perform
the MIC calculation for BMC management frames.

As discussed, we decided not to set IGTK to FW for consistency
between each cipher.
---
 mt7915/main.c | 4 ----
 1 file changed, 4 deletions(-)

--- a/mt7915/main.c
+++ b/mt7915/main.c
@@ -438,10 +438,6 @@ static int mt7915_set_key(struct ieee802
 
 	/* fall back to sw encryption for unsupported ciphers */
 	switch (key->cipher) {
-	case WLAN_CIPHER_SUITE_AES_CMAC:
-		wcid_keyidx = &wcid->hw_key_idx2;
-		key->flags |= IEEE80211_KEY_FLAG_GENERATE_MMIE;
-		break;
 	case WLAN_CIPHER_SUITE_TKIP:
 	case WLAN_CIPHER_SUITE_CCMP:
 	case WLAN_CIPHER_SUITE_CCMP_256:
